server/issue.go

```diff
@@ -760,17 +760,29 @@ func (s *Server) createPipelineFromIssue(ctx context.Context, issueCreate *api.I
 			}
 			// Convert to pipelineCreate
 			for i, stage := range p {
-				stageCreate := api.StageCreate{
-					Name: fmt.Sprintf("Deployment %v", i),
-				}
+				// Since environment is required for stage, we use an internal bb system environment for tenant deployments.
+				var environmentID *int
+				var taskCreateList []api.TaskCreate
 				for _, database := range stage {
+					if environmentID == nil {
+						environmentID = &database.Instance.EnvironmentID
+					}
+					if *environmentID != database.Instance.EnvironmentID {
+						err := fmt.Errorf("All databases in tenant mode deployment should have the same environment")
```

Comment:
To make this error more useful for user to debug, we should list all environments in this stage.

So we may need to write it in another way.

hasDifferentEnv : = false

if *environmentID != database.Instance.EnvironmentID {
hasDifferentEnv = true
break
}

if !hasDifferentEnv {
   error
}

Also would be more readable to include "Stage" in the error message. I am afraid deployment alone could be confusing since user may think it refers to the whole multi-stage deployment